### DISCUSSION TITLE
Allow "single result" APIs in autocomplete interface

### DISCUSSION THREAD

**Discussion by hermesalvesbr** [directus/directus#14633] [Feature Requests]
I'm trying to test with this api to get 'no array' values: 

https://api.zippopotam.us/us/33162

But, i can get just 'array values': 
```
{
    "post code": "33162",
    "country": "United States",
    "country abbreviation": "US",
    "places": [
        {
            "place name": "Miami",
            "longitude": "-80.183",
            "state": "Florida",
            "state abbreviation": "FL",
            "latitude": "25.9286"
        }
    ]
}
```

### To Reproduce

![image](https://user-images.githubusercontent.com/379850/180815990-ddb4442f-15c1-4c99-b05f-7baebf6bcb1a.png)

How can i get "country"? Because i can get just "places". 

I tried {}.country, country, {0}.country, [0].country in results path, text path, value path with or without and didn´t work. 

This is a bug or a limitation?

**Comment by rijkvanzanten** [comment]
The current implementation of the interface expects the API to return an array of results, so it can render the autocomplete dropdown section in the interface. Once the user clicks the desired result, the value under the `Value Path` is stored to the database. In this particular API, there are no "search results", so it's incompatible with the interface at this point. I've converted your bug report as a feature request discussion for this particular use case 👍🏻

**Reply by hermesalvesbr** [reply]
I need to create a custom endpoint to change the api for a result array? Or do you suggest another approach?

**Reply by rijkvanzanten** [reply]
What are you trying to achieve exactly? It sounds like you're asking the user to enter a zip code, and you're trying to save the place name? If that's the case, you might be better of setting up a regular text input for ZIP, and then using a flow to auto-download and populate read-only fields for place name etc 🤔

**Reply by hermesalvesbr** [reply]
I'm trying to get the country by the zip code.

**Reply by rijkvanzanten** [reply]
So you're trying to store the country in a new field? You should try that flow approach, I think you'll get the best of both worlds 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
